ruma-events/src/room/relationships.rs

```diff
@@ -0,0 +1,161 @@
+//! Types for event relationships.
+//!
+//! Events in Matrix can relate to one another in a couple of ways, this module
+//! adds types to parse the relationship of an event if any exists.
+//!
+//! MSC for all the relates_to types except replies:
+//!     https://github.com/matrix-org/matrix-doc/pull/2674
+
+use ruma_identifiers::EventId;
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+
+/// Enum modeling the different ways relationships can be expressed in a
+/// `m.relates_to` field of an event.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum RelatesTo {
+    /// A relation which contains subtypes indicating the type of the
+    /// relationship with the `rel_type` field.
+    Relation(Relation),
```

Comment:
This enum should not be nested. We should consider using the `Relation` name instead of `RelatesTo` though.

Reply:
Also, I understand why reactions can't be feature gated, but that doesn't apply here.